src/meltano/core/plugin/singer/tap.py

```diff
@@ -306,18 +306,30 @@ async def look_up_state(  # noqa: WPS231, WPS213, WPS210
 
             try:
                 shutil.copy(custom_state_path, state_path)
-                logger.info(f"Found state in {custom_state_filename}")  # noqa: G004
             except FileNotFoundError as err:
                 raise PluginExecutionError(
                     f"Could not find state file {custom_state_path}",  # noqa: EM102
                 ) from err
 
+            logger.info(f"Found state in {custom_state_filename}")  # noqa: G004
             return
-        # the `state.json` is stored in the database
-        if state := StateService(
+
+        # the `state.json` is stored in a state backend
+        state_service = StateService(
             project=elt_context.project,
             session=elt_context.session,
-        ).get_state(elt_context.job.job_name):
+        )
+        try:
+            state = state_service.get_state(elt_context.job.job_name)
+        except Exception as err:  # pragma: no cover
+            logger.error(
+                err.args[0],
+                state_backend=state_service.state_store_manager.label,
+            )
+            msg = "Failed to retrieve state"
+            raise PluginExecutionError(msg) from err
```

Comment:
This gives us a bit more context on why the tap failed to start:

```
2024-04-10T19:18:40.773990Z [error    ] The SSO session associated with this profile has expired or is otherwise invalid. To refresh this SSO session run aws sso login with the corresponding profile. state_backend=AWS S3
2024-04-10T19:18:40.777655Z [error    ] Cannot start plugin tap-getpocket: Failed to retrieve state
```